compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -249,7 +249,7 @@ trait QuotesAndSplices {
             val pat1 = if (patType eq patType1) pat else pat.withType(patType1)
             patBuf += pat1
           }
-        case Select(pat, _) if tree.symbol.isTypeSplice =>
+        case Select(pat: Bind, _) if tree.symbol.isTypeSplice =>
```

Comment:
- How are Select where the prefix is not a Bind handled?
- There are other cases which involve `pat` like `case Apply(fn, pat :: Nil)`, should those be `pat: Bind` too?

Reply:
> How are Select where the prefix is not a Bind handled?

Those just get the default `super.transform(tree)`. The `splitter` only needs to extract the bindings of the pattern. For types, these are the type variable inline definitions. If there is a `t.Underlying` that is not a pattern we assume that it is an explicit splice into the pattern. In case, `t` is not identified as a type variable because `t` is part of the path of the type. 

> There are other cases which involve pat like case Apply(fn, pat :: Nil), should those be pat: Bind too?

These are expression splices `${...}` in the pattern. There we can have an arbitrary pattern such as `${x}` or `${Expr(x)}`. Unlike with types, we do not have a notation or notion of inserting an expression is a quoted pattern.